dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ConvolutionalLayer.java

```diff
@@ -95,12 +174,25 @@ public Matrix feedForward(final Matrix input) {
    */
   @Override
   public Matrix backPropagate(final Matrix input, final Matrix activation, final Matrix nextError) {
-    throw new RuntimeException("Not Implemented");
+    final Matrix error = matrixFactory.create(input.getRows(), input.getColumns());
+    for (int n = 0; n < input.getColumns(); ++n) {
+      final Matrix im = col2im(nextError.getColumn(n).mmul(getLayerParameter().getWeightParam().transpose()));
+      error.putColumn(n, im);
+    }
+    return error;
   }
 
   /** {@inheritDoc} */
   @Override
   public LayerParameter generateParameterGradient(final Matrix input, final Matrix error) {
-    throw new RuntimeException("Not Implemented");
+    final Matrix weightDiff = matrixFactory.create(kernelHeight * kernelWidth, 1);
```

Comment:
Since we have not used a term "diff" for gradients, it would be better to use another term like "gradient". How about `weightGradient` instead of `weightDiff`?